src/plugins/chart_expressions/expression_xy/common/types/expression_functions.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { HorizontalAlignment, Position, VerticalAlignment } from '@elastic/charts';
+import { $Values } from '@kbn/utility-types';
+import { Datatable } from '../../../../expressions';
+import { PaletteOutput } from '../../../../charts/common';
+import { EventAnnotationOutput } from '../../../../event_annotation/common';
+import {
+  AxisExtentModes,
+  FillStyles,
+  FittingFunctions,
+  IconPositions,
+  LayerTypes,
+  MULTITABLE,
+  LineStyles,
+  SeriesTypes,
+  ValueLabelModes,
+  XScaleTypes,
+  XYCurveTypes,
+  YAxisModes,
+  YScaleTypes,
+  REFERENCE_LINE_LAYER,
+  Y_CONFIG,
+  AXIS_TITLES_VISIBILITY_CONFIG,
+  LABELS_ORIENTATION_CONFIG,
+  TICK_LABELS_CONFIG,
+  GRID_LINES_CONFIG,
+  LEGEND_CONFIG,
+  DATA_LAYER,
+  AXIS_EXTENT_CONFIG,
+  ANNOTATION_LAYER,
+  EndValues,
+} from '../constants';
+
+export type EndValue = $Values<typeof EndValues>;
```

Comment:
out of curiosity - is `$Values<typeof EndValues>` the same as `keyof typeof EndValues`? Not suggesting the change, just curious because I never used it :)

Reply:
gotcha, thanks! 👌🏼